dlp/snippets/deid.py

```diff
@@ -1581,6 +1581,73 @@ def deidentify_table_replace_with_info_types(
 # [END dlp_deidentify_table_infotypes]
 
 
+# [START dlp_deidentify_dictionary_replacement]
+def deindentify_with_dictionary_replacement(
+    project: str,
+    input_str: str,
+    info_types: List[str],
+    word_list: List[str],
+) -> None:
+
+    """Uses the Data Loss Prevention API to de-identify sensitive data in a
+    string by replacing each piece of detected sensitive data with a value
+    that Cloud DLP randomly selects from a list of words that you provide.
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        input_str: The string to deidentify (will be treated as text).
+        info_types: A list of strings representing info types to look for.
+        word_list: List of words or phrases to search for in the data.
+    """
+
+    # Import the client library
+    import google.cloud.dlp
+
+    # Instantiate a client
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Construct inspect configuration dictionary
+    inspect_config = {
+        "info_types": [{"name": info_type} for info_type in info_types]
+    }
+
+    # Construct deidentify configuration dictionary
+    deidentify_config = {
+        "info_type_transformations": {
+            "transformations": [
+                {
```

Comment:
The json sample provided recommends setting infotype in the transformations as well.
`
"deidentifyConfig": {
    "infoTypeTransformations": {
      "transformations": [
        {
          "infoTypes": [
            {
              "name": "EMAIL_ADDRESS"
            }
          ],
          "primitiveTransformation": {
            "replaceDictionaryConfig": {
              "wordList": {
                "words": [
                  "izumi@example.com",
                  "alex@example.com",
                  "tal@example.com"
                ]
              }
            }
          }
        }
      ]
    }
  }`

Reply:
yeah , for python,  declaring info types as part of inspect config and not as part of transformations is generally followed for all deid samples. Eg: deidentify_with_mask, deidentify_with_redact etc.

Reply:
Please note that this an API related and not language specific. It is not declaration if infoType, it is expressing expected behaviour.  In the current form we are riding on default behaviour. It matter when the user wants different transformations for different infotypes (more complex scenarios). The examples should provide correct usage guidance.

Reply:
I understand that. Updated now, please check.